scipy/special/special_c_wrappers.cpp

```diff
@@ -0,0 +1,204 @@
+/* These wrappers exist to allow for calling functions from the under development
+ * header only C++ special function library in scipy/special/special within files
+ * for special functions which are still implemented in C or Cython. The hope is
+ * that these  wrappers will become unnecessary if all of special is translated
+ * into C++ */
+
+extern "C" {
+
+#include <numpy/npy_math.h>
+#include "special_c_wrappers.h"
+}
+
+#include "special/binom.h"
+#include "special/hyp2f1.h"
+#include "special/cephes/airy.h"
+#include "special/cephes/beta.h"
+#include "special/cephes/bdtr.h"
+#include "special/cephes/ellpj.h"
+#include "special/cephes/ellpk.h"
+#include "special/cephes/expn.h"
+#include "special/cephes/hyp2f1.h"
+#include "special/cephes/fresnl.h"
+#include "special/cephes/gamma.h"
+#include "special/cephes/scipy_iv.h"
+#include "special/cephes/jv.h"
+#include "special/cephes/kolmogorov.h"
+#include "special/cephes/nbdtr.h"
+#include "special/cephes/ndtr.h"
+#include "special/cephes/ndtri.h"
+#include "special/cephes/pdtr.h"
+#include "special/cephes/poch.h"
+#include "special/cephes/polevl.h"
+#include "special/cephes/sici.h"
+#include "special/cephes/shichi.h"
+#include "special/cephes/struve.h"
+#include "special/cephes/unity.h"
+#include "special/cephes/yn.h"
+
+
+extern "C" double binom_wrap(double n, double k) {
```

Comment:
I think, but am not certain, that if the prototypes in the header are declared `extern "C"` we don't need to duplicate it in the `.cpp` file.